torchvision/datasets/_stereo_matching.py

```diff
@@ -0,0 +1,1063 @@
+import json
+import os
+import random
+import re
+import shutil
+import warnings
+from abc import ABC, abstractmethod
+from glob import glob
+from pathlib import Path
+from typing import Callable, List, Optional, Tuple
+
+import numpy as np
+from PIL import Image
+from torch import Tensor
+
+from .utils import download_and_extract_archive, verify_str_arg
+from .vision import VisionDataset
+
+__all__ = (
+    "CREStereo"
+    "StereoMiddlebury2014"
+    "StereoETH3D"
+    "StereoKitti2012"
+    "StereoKitti2015"
+    "StereoSintel"
+    "StereoSceneFlow"
+    "StereoFallingThings"
+    "InStereo2k"
+)
+
+
+def read_pfm_file(file_path: str) -> np.array:
```

Comment:
Could we use or modify our existing https://github.com/pytorch/vision/blob/main/torchvision/datasets/_optical_flow.py#L477:L477 for this?

Reply:
Well, technically based on the header file information, .pfm files could store 1 or 3 channels for (H, W) information. During some initial tests I saw that most usually it was the case that they were storing just 1 channel of information, so the reshape in _optical_flow should be dynamic.

There's also the data slicing. Technically, if correct data is provided and the correct are files are linked to the dataset slicing an extra channel `data[:2, :, :]` would yield no issues. However, the default that exists in the test utils [make_pfm_file](https://github.com/pytorch/vision/blob/9effc4cdc940952500c72b3ea5a0d7348f33b004/test/datasets_utils.py#L934) implicitly creates .pfm files with 3 channels.

```python3
data = data.reshape(h, w, 3).transpose(2, 0, 1) # <--- move to something like data.reshape(h, w, c)
data = np.flip(data, axis=1)  # flip on h dimension
data = data[:2, :, :]
return data.astype(np.float32)
```

Reply:
It seems that we can modify the existing then, if so that would be preferable.
If you go down this path, please do the change in a separate PR. @NicolasHug would it make sense to move the util out of the optical_flow module?

Reply:
As discussed offline with @TeodorPoncu we decided to move the `read_pfm_file` to `datasets.utils` in a separate PR and add an extra argument in a BC way to handle the different scenarios. Would you agree with this approach @NicolasHug ?

Reply:
sure

Reply:
lol didn't see your comment and basically suggested what you did :D